frappe/core/doctype/file/file.py

```diff
@@ -663,6 +663,15 @@ def get_web_image(file_url):
 	extn = get_extension(filename, extn, r.content)
 	filename = "/files/" + strip(unquote(filename))
 
+	# Get image extension from response object 'r'
+	if not extn:
+		content_type = r.headers.get("Content-Type") 
+		
+		if content_type:
+			from mimetypes import guess_extension
+
+			extn = guess_extension(content_type)[1:]
```

Comment:
lets do both solutions?
- Guess extension from content type
- If still not available then save it without extension.


It should be possible to save images without extension too! Browsers don't care about it when rendering images.

Reply:
I thought this logic belonged in the `guess_extension` util; raised an alt fix for this to `develop`